pkg/plan/mock/generator.go

```diff
@@ -0,0 +1,192 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package planmock
+
+import (
+	"math/rand"
+
+	"github.com/elastic/cloud-sdk-go/pkg/api/mock"
+	"github.com/elastic/cloud-sdk-go/pkg/models"
+	"github.com/elastic/cloud-sdk-go/pkg/util/ec"
+)
+
+const letterBytes = "abcdefghijklmnopqrstuvwxyz1234567890"
+
+// GenerateConfig is a helper used to create a DeploymentGetResponse to be
+// used for tests. It provides a generic implementation so all deployment
+// resoource types can be generated a plan log and tracked.
+type GenerateConfig struct {
+	// If omitted, a random ID will be auto-generated.
+	ID string
+
+	Elasticsearch []GeneratedResourceConfig
+	Kibana        []GeneratedResourceConfig
+	Apm           []GeneratedResourceConfig
+	Appsearch     []GeneratedResourceConfig
+}
+
+// GeneratedResourceConfig is used to construct a deployment resource plan
+// with the plan log steps specified.
+type GeneratedResourceConfig struct {
+	// If omitted, a random ID will be auto-generated.
+	ID string
+
+	// If omitted, it'll default to "main-<type>"
```

Comment:
I think that's correct, I would keep the Go API close to our HTTP API. Probably ecctl should also have a `--kind` flag rather than a `--type` flag. Probably good idea to do that before we finish the _beta_ label.